Model/AdminNotification.php

```diff
@@ -78,4 +110,28 @@ public function send($customer)
 
         $this->transportBuilder->getTransport()->sendMessage();
     }
+
+    private function getCustomerData(CustomerInterface $customer): array
+    {
+        $customerData = $this->dataObjectProcessor->buildOutputDataArray($customer, CustomerInterface::class);
+        $customerData['name'] = $this->customerViewHelper->getCustomerName($customer);
+
+        return $customerData;
+    }
+
+    private function getFormattedShippingAddress(CustomerInterface $customer): ?string
+    {
+        if ($customer->getDefaultShipping() === null) {
+            return null;
```

Comment:
Isn't the default shipping address almost always null because the customer never had the chance to set one before they are registered and activated? Or am I missing something?

Reply:
You can require the customer to enter address details upon registration (by setting customer_form_register to true for the block `customer_form_register`, see https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Customer/view/frontend/templates/form/register.phtml#L76) and this address is then set as default billing and shipping address (see https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Customer/view/frontend/templates/form/register.phtml#L224-L225).

So the idea is if you enable the address in the register form you can `formattedShippingAddress` in the email template.

Reply:
Ah okay, thanks for the explanation!